object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -490,16 +501,21 @@ private void handleAssignFromRHS(
         LocalVariableNode mustCallChoiceParamLocal = getLocalPassedAsMustCallChoiceParam(rhs);
         if (mustCallChoiceParamLocal != null) {
           if (isVarInDefs(newDefs, mustCallChoiceParamLocal)) {
-            ImmutableSet<LocalVarWithTree> immutableSet =
+            LocalVarWithTree latestAssignmentPair =
+                getAssignmentTreeOfVar(newDefs, (LocalVariableNode) lhs);
+
+            ImmutableSet<LocalVarWithTree> setContainingLatestAssignmentPair =
                 getSetContainingAssignmentTreeOfVar(newDefs, mustCallChoiceParamLocal);
-            ImmutableSet<LocalVarWithTree> newImmutableSet =
-                FluentIterable.from(immutableSet)
+            ImmutableSet<LocalVarWithTree> newSetContainingLatestAssignmentPair =
+                FluentIterable.from(setContainingLatestAssignmentPair)
+                    .filter(Predicates.not(Predicates.equalTo(latestAssignmentPair)))
                     .append(
                         new LocalVarWithTree(
                             new LocalVariable((LocalVariableNode) lhs), node.getTree()))
                     .toSet();
-            newDefs.remove(immutableSet);
-            newDefs.add(newImmutableSet);
+
+            newDefs.remove(setContainingLatestAssignmentPair);
+            newDefs.add(newSetContainingLatestAssignmentPair);
           }
         } else {
```

Comment:
Let's add a check under the `else` that `lhs` is not already part of `newDefs`, and if it is, throw a `RuntimeException` (to ensure our logic is correct)

Reply:
We shouldn't throw an Exception here! If it's not part of `newDefs`, it means we shouldn't track this resource (e.g. this resource is MCC with `@NotOwning` parameters)!

Reply:
Sorry, I meant we should throw an exception if `lhs` _is_ already in `newDefs`.  I think that should be impossible under this `else` case, which is why the code just adds the new `LocalVarWithTree` for `lhs` without checking if it's already there.  Or did I miss something?

Reply:
Sorry, I didn't read it carefully. 
https://github.com/kelloggm/object-construction-checker/pull/282/commits/171817635c00a3c0b87306ba2feb7fae90f9bb92